CONTRIBUTING.md

```diff
@@ -16,3 +18,36 @@ Prometheus uses GitHub to manage reviews of pull requests.
   and the _Formatting and style_ section of Peter Bourgon's [Go: Best
   Practices for Production
   Environments](http://peter.bourgon.org/go-in-production/#formatting-and-style).
+
+
+## Steps to Contribute
+
+Should you wish to work on an issue, please claim it first by commenting on the GitHub issue that you want to work on it. This is to prevent duplicated efforts from contributors on the same issue.
+
+Please check the [`low-hanging-fruit`](https://github.com/prometheus/prometheus/issues?q=is%3Aissue+is%3Aopen+label%3A%22low+hanging+fruit%22) label to find issues that are good for getting started. If you have questions about one of the issues, with or without the tag, please comment on them and one of the maintainers will clarify it. For a quicker response, contact us over [IRC](https://prometheus.io/community).
+
+For complete instructions on how to compile see: [Building From Source](https://github.com/prometheus/prometheus#building-from-source)
+
+For quickly compiling and testing your changes do:
+```
+# For building.
+go build ./cmd/prometheus/
+./prometheus
+
+# For testing.
+make test-short   # To quickly test.
+
+make test         # For more thorough testing before you commit and push.
```

Comment:
Is it really worth distinguishing here?
While both exist, I think we never properly defined what makes a test short or long and running everything via `make test` just takes a few seconds. We should probably remove the short thing for now and only add it back once we have long running stuff like SD integration.